metal/mmtl/dataset.py

```diff
@@ -84,14 +84,14 @@ def __init__(
                 True: (tokens, segment), labels
                 False: tokens, labels
         """
-        tokenizer = BertTokenizer.from_pretrained(bert_model, do_lower_case=True)
+        self.tokenizer = BertTokenizer.from_pretrained(bert_model, do_lower_case=True)
```

Comment:
why do we need to assign to self here?

Reply:
Because we need it to be able to take the dataset and re-compute new labels for it when the new task (e.g. BLEU score) requires the original tokens -- we need the inds_to_tokens method to get the raw strings back, b/c lots of useful NLP utilities (e.g. NLTK) need the raw tokens.